src/main/java/org/apache/mesos/hdfs/Scheduler.java

```diff
@@ -108,16 +246,16 @@ private void launchNode(SchedulerDriver driver, Offer offer,
           .build();
       tasks.add(task);
 
-      liveState.addStagingTask(taskId);
-      liveState.addTask(taskId, offer.getHostname(), offer.getSlaveId().getValue());
+      liveState.addStagingTask(task);
+      persistentState.addNode(taskId, offer.getHostname(), taskName);
```

Comment:
`addNode` sounds too vague, and makes me think you're adding a Mesos node, not necessarily an HDFS Node task (on a particular host). Maybe "addHdfsNode" or "add[Scheduled/Launched/Placed]Task"?